libs/langchain-ollama/src/tests/chat_models_think.int.test.ts

```diff
@@ -52,3 +52,175 @@ test("test deep seek model with think=true (default)", async () => {
   expect(responseContent).toMatch(/photosynthesis/i); // Check it includes the topic
   expect(responseContent.length).toBeGreaterThan(1);
 });
+
+test("test type safety for thinking parameter values", async () => {
+  // Test that TypeScript accepts all valid string and boolean values
+  const ollamaHigh = new ChatOllama({
+    model: "deepseek-r1:32b",
+    think: "high" as const,
+    maxRetries: 1,
+  });
+  
+  const ollamaMedium = new ChatOllama({
+    model: "deepseek-r1:32b", 
+    think: "medium" as const,
+    maxRetries: 1,
+  });
+  
+  const ollamaLow = new ChatOllama({
+    model: "deepseek-r1:32b",
+    think: "low" as const,
+    maxRetries: 1,
+  });
+  
+  const ollamaTrue = new ChatOllama({
+    model: "deepseek-r1:32b",
+    think: true,
+    maxRetries: 1,
+  });
+  
+  const ollamaFalse = new ChatOllama({
+    model: "deepseek-r1:32b",
+    think: false,
+    maxRetries: 1,
+  });
+
+  // All should be properly instantiated
+  expect(ollamaHigh).toBeDefined();
+  expect(ollamaMedium).toBeDefined();
+  expect(ollamaLow).toBeDefined();
+  expect(ollamaTrue).toBeDefined();
+  expect(ollamaFalse).toBeDefined();
+  
+  // Quick test that string values work in practice
+  const res = await ollamaHigh.invoke([
+    new HumanMessage({ content: "How many r in the word strawberry?" })
+  ]);
+  
+  expect(res).toBeDefined();
+  expect(typeof res.content).toBe("string");
+  expect(res.content.length).toBeGreaterThan(0);
+});
+
+test.each([
+  { thinkLevel: "high" as const },
+  { thinkLevel: "medium" as const },
+  { thinkLevel: "low" as const },
+])("test string thinking parameter '$thinkLevel'", async ({ thinkLevel }) => {
+  const ollama = new ChatOllama({
+    model: "deepseek-r1:32b",
+    think: thinkLevel,
+    maxRetries: 1,
+  });
+
+  const res = await ollama.invoke([
+    new HumanMessage({
+      content: "How many r in the word strawberry?",
+    }),
+  ]);
+
+  // Ensure the response is defined
+  expect(res).toBeDefined();
+  expect(res.content).toBeDefined();
+  expect(res.additional_kwargs).toBeDefined();
+
+  // Validate basic functionality - response should exist and not contain thinking tags
+  expect(typeof res.content).toBe("string");
+  expect(res.content.length).toBeGreaterThan(0);
+  expect(res.content).not.toMatch(/<think>.*?<\/think>/is); // No thinking tags in content
+
+  // For string thinking levels, validate thinking content if present
+  if (res.additional_kwargs?.thinking_content) {
```

Comment:
Why the `if` statement? Shouldn't we expect this flag to be set? I am afraid if we introduce a bug were this becomes `false` for some reason , we stop doing all the assertions and may miss a bug.